code/modules/mod/mod.dm

```diff
@@ -0,0 +1,555 @@
+		mod_parts += gauntlets
+	else
+		CRASH("A MODsuit spawned without gauntlets.")
+	if(ispath(theme.boots_path))
+		boots = new theme.boots_path(src)
+		boots.mod = src
+		mod_parts += boots
+	else
+		CRASH("A MODsuit spawned without boots.")
+	var/list/all_parts = mod_parts.Copy() + src
+	for(var/obj/item/piece as anything in all_parts)
+		piece.name = "[theme.name] [piece.name]"
+		piece.desc = "[piece.desc] [theme.desc]"
+		piece.armor = getArmor(arglist(theme.armor))
+		piece.resistance_flags = theme.resistance_flags
+		piece.max_heat_protection_temperature = theme.max_heat_protection_temperature
+		piece.min_cold_protection_temperature = theme.min_cold_protection_temperature
+		piece.gas_transfer_coefficient = theme.gas_transfer_coefficient
+		piece.permeability_coefficient = theme.permeability_coefficient
+		piece.siemens_coefficient = theme.siemens_coefficient
+		piece.icon_state = "[skin]-[initial(piece.icon_state)]"
+	if(initial_modules.len)
+		for(var/obj/item/mod/module/module in initial_modules)
+			module = new module(src)
+			install(module, TRUE)
+	RegisterSignal(src, COMSIG_ATOM_EXITED, .proc/on_exit)
+	movedelay = CONFIG_GET(number/movedelay/run_delay)
+
+/obj/item/mod/control/Destroy()
+	STOP_PROCESSING(SSobj, src)
+	QDEL_NULL(wires)
+	QDEL_NULL(cell)
+	if(helmet)
+		helmet.mod = null
+		QDEL_NULL(helmet)
+	if(chestplate)
+		chestplate.mod = null
+		QDEL_NULL(chestplate)
+	if(gauntlets)
+		gauntlets.mod = null
+		QDEL_NULL(gauntlets)
+	if(boots)
+		boots.mod = null
+		QDEL_NULL(boots)
+	for(var/obj/item/mod/module/module in modules)
+		module.mod = null
+		QDEL_NULL(module)
+	..()
+
+/obj/item/mod/control/process(delta_time)
+	if(seconds_electrified > MACHINE_NOT_ELECTRIFIED)
+		seconds_electrified--
+	if(!cell?.charge && active && !activating)
+		power_off()
+		return PROCESS_KILL
+	var/malfunctioning_charge_drain = 0
+	if(malfunctioning)
+		malfunctioning_charge_drain = rand(1,20)
+	cell.charge = max(0, cell.charge - (cell_drain + malfunctioning_charge_drain)*delta_time)
+	for(var/obj/item/mod/module/module in modules)
+		if(malfunctioning && module.active && DT_PROB(5, delta_time))
+			module.on_deactivation()
+		module.on_process(delta_time)
+
+/obj/item/mod/control/equipped(mob/user, slot)
+	..()
+	if(slot == ITEM_SLOT_BACK)
+		wearer = user
+		RegisterSignal(wearer, COMSIG_ATOM_EXITED, .proc/on_exit)
+	else if(wearer)
+		UnregisterSignal(wearer, COMSIG_ATOM_EXITED)
+		wearer = null
+
+/obj/item/mod/control/dropped(mob/user)
+	..()
+	wearer = null
+
+/obj/item/mod/control/item_action_slot_check(slot)
+	if(slot == ITEM_SLOT_BACK)
+		return TRUE
+
+/obj/item/mod/control/allow_attack_hand_drop(mob/user)
+	var/mob/living/carbon/carbon_user = user
+	if(!istype(carbon_user) || src != carbon_user.back)
+		return ..()
+	for(var/obj/item/part in mod_parts)
+		if(part.loc != src)
+			to_chat(carbon_user, "<span class='warning'>ERROR: At least one of the parts are still on your body, please retract them and try again.</span>")
```

Comment:
is it possible to make this try to retract the parts first?

Reply:
no it physicaly impsosible